wasm-test/src/lib.rs

```diff
@@ -1,21 +1,41 @@
-
+#[macro_use]
 extern crate hdk;
 extern crate holochain_wasm_utils;
+extern crate serde;
+extern crate serde_json;
+#[macro_use]
+extern crate serde_derive;
 
+use hdk::globals::G_MEM_STACK;
 use holochain_wasm_utils::*;
-use hdk::globals::g_mem_stack;
 
 #[no_mangle]
-pub extern "C" fn check_global_dispatch(encoded_allocation_of_input : u32) -> u32 {
-  unsafe {
-    g_mem_stack = Some(SinglePageStack::new_from_encoded(encoded_allocation_of_input));
-  }
+pub extern "C" fn check_global(encoded_allocation_of_input: u32) -> u32 {
+    unsafe {
+        G_MEM_STACK = Some(SinglePageStack::new_from_encoded(
+            encoded_allocation_of_input,
+        ));
+    }
+
     hdk::debug(&hdk::APP_NAME);
     hdk::debug(&hdk::APP_DNA_HASH);
     hdk::debug(&hdk::APP_AGENT_ID_STR);
     hdk::debug(&hdk::APP_AGENT_KEY_HASH);
     hdk::debug(&hdk::APP_AGENT_INITIAL_HASH);
     hdk::debug(&hdk::APP_AGENT_LATEST_HASH);
 
-  return 0;
+    return 0;
+}
+
+#[derive(Serialize)]
+struct TweetResponse {
+    ok: bool,
+}
+
+zome_functions! {
```

Comment:
So all Zome functions will have to be defined like this?

Reply:
Yes. All of the top level zome functions will be written inside the macro